lib/src/emitter.dart

```diff
@@ -535,6 +536,38 @@ class DartEmitter extends Object
     return out;
   }
 
+  @override
+  StringSink visitRecordType(RecordType spec, [StringSink? output]) {
+    final out = (output ??= StringBuffer())..write('(');
+    visitAll<Reference>(spec.positionalFieldTypes, out, (spec) {
+      spec.accept(this, out);
+    });
+    if (spec.namedFieldTypes.isNotEmpty) {
+      if (spec.positionalFieldTypes.isNotEmpty) {
+        out.write(', ');
+      }
+      out.write('{');
+      visitAll<MapEntry<String, Reference>>(spec.namedFieldTypes.entries, out,
+          (entry) {
+        entry.value.accept(this, out);
+        out.write(' ${entry.key}');
+      });
+      out.write('}');
+    } else {
+      if (spec.positionalFieldTypes.length == 1) {
```

Comment:
[nit] can we use `} else if (` to remove 1 level of indentation like you did in `expression.dart`